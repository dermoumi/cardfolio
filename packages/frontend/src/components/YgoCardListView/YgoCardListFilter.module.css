.filter-bar {
  flex: 0 0 auto;
  background-color: #fff;
  display: flex;
  align-items: stretch;

  & > * {
    flex: 1;
  }
}
