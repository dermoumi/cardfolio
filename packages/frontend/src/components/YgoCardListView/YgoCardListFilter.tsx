import type { YgoCardFilters } from "@/types";
import type { FC } from "react";

import { TextInput } from "@cardfolio/ui";
import { useCallback } from "react";

import styles from "./YgoCardListFilter.module.css";

export type YgoCardListFilterProps = {
  filters: YgoCardFilters;
  setFilters: (filter: YgoCardFilters) => void;
};

const YgoCardListFilter: FC<YgoCardListFilterProps> = ({ filters, setFilters }) => {
  const setNameFilter = useCallback((name: string) => {
    setFilters({ ...filters, name });
  }, [setFilters, filters]);

  return (
    <div className={styles.filterBar}>
      <TextInput
        name="card-name"
        type="search"
        placeholder="Search cards..."
        value={filters.name || ""}
        onChange={setNameFilter}
      />
    </div>
  );
};

export default YgoCardListFilter;
