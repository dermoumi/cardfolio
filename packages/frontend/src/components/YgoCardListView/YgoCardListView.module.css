.card-list-view {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  height: 100%;
}

.grid-container {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  flex: 1 1 auto;
  overflow: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
  padding: 0.2rem;
  background: #fafafa;
}
