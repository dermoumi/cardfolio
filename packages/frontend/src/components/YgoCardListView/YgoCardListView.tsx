import type { YgoCard, YgoCardFilters } from "@/types";
import type { FC } from "react";

import { Button } from "@cardfolio/ui";

import YgoCardGrid from "../YgoCardGrid";
import YgoCardListFilter from "./YgoCardListFilter";
import styles from "./YgoCardListView.module.css";

export type YgoCardListProps = {
  cards: Array<YgoCard>;
  fetchNextPage: () => void;
  hasNextPage?: boolean;
  isFetchingNextPage?: boolean;
  filters: YgoCardFilters;
  setFilters: (filter: YgoCardFilters) => void;
};

const YgoCardListView: FC<YgoCardListProps> = (
  { cards, fetchNextPage, hasNextPage, isFetchingNextPage, filters, setFilters },
) => {
  return (
    <div className={styles.cardListView}>
      <YgoCardListFilter filters={filters} setFilters={setFilters} />
      <div className={styles.gridContainer}>
        <YgoCardGrid cards={cards} />
        {hasNextPage && (
          <Button onClick={() => fetchNextPage()} disabled={isFetchingNextPage}>
            {isFetchingNextPage ? "Loading..." : "Load More"}
          </Button>
        )}
      </div>
    </div>
  );
};

export default YgoCardListView;
