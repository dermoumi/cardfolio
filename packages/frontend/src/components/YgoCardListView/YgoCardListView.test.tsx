import type { YgoCard } from "@/types";

import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import YgoCardListView from ".";

describe("YgoCardListView", () => {
  const mockCards: Array<YgoCard> = [
    {
      "id": 15,
      "updatedAt": "2025-09-03T21:54:04.817564Z",
      "name": "7 Colored Fish",
      "description": "A rare rainbow fish that has never been caught by mortal man.",
      "kind": "monster",
      "password": "23771716",
      "konamiId": 4446,
      "tcgDate": "2002-06-26",
      "ocgDate": "2000-01-27",
      "monsterKind": "normal",
      "monsterAttribute": "water",
      "monsterRace": "fish",
      "monsterAtk": 1800,
      "monsterDef": 800,
      "monsterLevel": 4,
    },
    {
      "id": 10,
      "updatedAt": "2025-09-03T21:54:02.391850Z",
      "name": "1st Movement Solo",
      "description":
        "If you control no monsters: Special Summon 1 Level 4 or lower \"Melodious\" monster from your hand or Deck. You can only activate 1 \"1st Movement Solo\" per turn. You cannot Special Summon monsters during the turn you activate this card, except \"Melodious\" monsters.",
      "kind": "spell",
      "password": "44256816",
      "konamiId": 11391,
      "tcgDate": "2014-11-06",
      "ocgDate": "2014-07-19",
      "spellKind": "normal",
    },
  ];

  it("matches snapshot", () => {
    const { asFragment } = render(
      <YgoCardListView
        cards={mockCards}
        fetchNextPage={vi.fn()}
        hasNextPage={false}
        isFetchingNextPage={false}
        filters={{ name: "test" }}
        setFilters={vi.fn()}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
