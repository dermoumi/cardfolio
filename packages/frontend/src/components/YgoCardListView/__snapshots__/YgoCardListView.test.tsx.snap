// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`YgoCardListView > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_cardListView_177be7"
  >
    <div
      class="_filterBar_4db3c9"
    >
      <input
        class="_textInput_5ec5c9"
        name="card-name"
        placeholder="Search cards..."
        type="search"
        value="test"
      />
    </div>
    <div
      class="_gridContainer_177be7"
    >
      <div
        class="_ygoCardGrid_898a22"
      >
        <div
          class="_ygoCardItem_b4a956"
        >
          <img
            alt="7 Colored Fish"
            src="/api/v1/ygo/cards/15/image"
          />
        </div>
        <div
          class="_ygoCardItem_b4a956"
        >
          <img
            alt="1st Movement Solo"
            src="/api/v1/ygo/cards/10/image"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
